test/unit/CSV/CSVWriterTest.php

```diff
@@ -8,45 +8,55 @@
 
 
 use Ingenerator\PHPUtils\CSV\CSVWriter;
+use PHPUnit\Framework\TestCase;
 
-class CSVWriterTest extends \PHPUnit_Framework_TestCase
+class CSVWriterTest extends TestCase
 {
 
+    protected $newSubject;
+
+    protected function setUp()
+    {
+        $this->newSubject = new CSVWriter;
+    }
+
     public function test_it_is_initialisable()
     {
-        $this->assertInstanceOf(CSVWriter::class, $this->newSubject());
+        $this->assertInstanceOf(CSVWriter::class, $this->newSubject);
     }
 
     /**
-     * @expectedException \RuntimeException
+     * @expectedException \ErrorException
      */
     public function test_it_throws_if_file_cannot_be_opened()
     {
-        $this->markTestIncomplete();
+        $this->newSubject->open('/invalid_csv_file');
     }
 
     /**
      * @expectedException \LogicException
      */
     public function test_it_throws_if_writing_before_opening_file()
     {
-        $this->newSubject()->write(['any' => 'junk']);
+        $this->newSubject->write(['any' => 'junk']);
     }
 
+    /**
+     * @expectedException \LogicException
+     */
     public function test_it_throws_if_writing_after_closing_file()
     {
-        $subj = $this->newSubject();
+        $subj = $this->newSubject;
         $subj->open('php://temp');
         $subj->close();
-        $this->setExpectedException(\LogicException::class);
```

Comment:
This was deliberately here instead of as an `@expectedException` to make sure that it only passes if the exception is actually thrown in write() - if it's thrown from newSubject/open/close the test should fail.

If that's not clear, you could add a comment, or do it another way e.g.
```php
try {
  $subj = $this->newSubject();
  $subj->open('php://temp');
  $subj->close();
} catch (\Exception $e) {
  $this->fail('Unexpected exception '.$e);
}
$subj->write(['any' => 'content']);
```

Reply:
I will add this ```try...catch``` block to be another way.